pypdf/_cmap.py

```diff
@@ -75,28 +76,28 @@ def build_char_map_from_dict(
         for x in int_entry:
             if x <= 255:
                 encoding[x] = chr(x)
-    try:
-        # override space_width with new params
-        space_width = _default_fonts_space_width[cast(str, ft["/BaseFont"])]
-    except Exception:
-        pass
     # I consider the space_code is available on one byte
     if isinstance(space_code, str):
         try:  # one byte
             sp = space_code.encode("charmap")[0]
         except Exception:
             sp = space_code.encode("utf-16-be")
             sp = sp[0] + 256 * sp[1]
+        try:
+            sp = ord(map_dict[chr(sp)])
+        except Exception:
```

Comment:
Can we make this exception more specific? I would assume that `KeyError` would be possible, but others?

Reply:
Thanks, I noticed the bug!